resources/js/beatmap-discussions/user-filter.tsx

```diff
@@ -62,27 +82,34 @@ export class UserFilter extends React.Component<Props> {
     );
   }
 
+  private getGroup(option: Option) {
+    if (this.isOwner(option)) return mapperGroup;
+    if (option.groups == null || option.groups.length === 0) return null;
```

Comment:
I wonder if `(!x)` can be used instead of `(x == null || x.length === 0)`, typescript narrows the type down correctly and this pattern appears in a few places 🤔

Reply:
I don't think the `strict-boolean-expressions` lint will be happy with that

Reply:
also the function above is more than just that pattern - it'll allow accessing any index while still being mobx compatible

Reply:
`strict-boolean-expressions` is fine with it with the current setting; apparently it's also "safe" https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md#allownumber

Reply:
`allowNullableNumber` isn't safe though but this is also object and number 🤔